partcad/src/partcad/part_factory_enrich.py

```
@@ -19,50 +19,62 @@
-            super().__init__(ctx, source_project, target_project, config)
-
             # Determine the part the 'enrich' points to
             if "source" in config:
-                self.source_part_name = config["source"]
+                source_part_name = config["source"]
             else:
-                self.source_part_name = config["name"]
-                if not "project" in config:
+                source_part_name = config["name"]
+                if "project" not in config:
                     raise Exception("Enrich needs either the source part name or the source project name")
```

Ruff (0.8.2)

Create your own exception

(TRY002)